### PR TITLE
Allow programmatic config changes

### PR DESCRIPTION
**Pull request by prodion23**
## Description
This allows programatic configuration changes.
```
agent = Agent()
with agent.edit_config() as config:
   config.service_name = "service name set programatically"
```

### Testing
- updated `test_flask_1.py` inspects span resources to check that the service name has been updated
- tested manually with both manual instrumentation & auto instrumentation

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by prodion23** [comment]
Pipeline is failing because of an unlocked dep version. 
I spent some time trying to figure it out but will have to come back to it after bit.  

Aside from this I think the general approach of the pr is okay to review.

File: README.md [hunk 1]
```diff
@@ -54,14 +61,23 @@ hypertrace-instrument python app.py
...
 
+In some scenarios you may want to configure your agent options programmatically, the following example shows how to achieve this. 
+```python
+from hypertrace.agent import Agent
+
+agent = Agent()
+a = Agent()
+with a.edit_config() as config:
+    config.service_name = "code modified service name"
+    config.reporting.endpoint = "http://localhost:9411/api/v2/spans"
```
Comment (davexroth) [inline README.md:78]: can you use OTLP as the example, and specify `trace_reporter_type` also

File: tests/postgresql/tox.ini [hunk 2]
```diff
@@ -31,7 +31,9 @@ commands =
     docker-compose pull db
     bash -ec 'if [ `uname` = '"'"'Linux'"'"' ] && [ "{env:GITHUB_ACTIONS:false}" != true ]; then chcon -h system_u:object_r:bin_t:s0 sql; chcon -Rt svirt_sandbox_file_t sql; fi'
     docker-compose up -d db
-    sleep 20
+    sleep 30
+    pip install psycopg2-binary==2.8.6
```
Comment (rcbjBlueMars) [inline tests/postgresql/tox.ini:35]: These modules and specific versions should be referenced in requirements.txt in this directory. Not explictly installed in tox.ini.

File: tests/flask/test_flask_1.py [hunk 3]
```diff
@@ -1,191 +1,195 @@
 import sys
```
[the remark this answers is not in the document]
Comment (prodion23) [reply tests/flask/test_flask_1.py:1]: Reverted flask test change, added new test file for this

File: tests/postgresql/tox.ini (hunk 2 above)
Comment (prodion23) [reply tests/postgresql/tox.ini]: I agree - something related to postgres isn't version locked somewhere so this was just an attempt to fix it closest to the failing test.  

Luckily(at least in regard to this pr) - it's also failing on latest main changes:
https://github.com/hypertrace/pythonagent/pull/229

Will update the pr once the issue is sorted

**Review by tim-mwangi — APPROVED** [review]

**Pull request merged**